invenio_accounts/models.py

```diff
@@ -72,6 +76,22 @@ def __str__(self):
     """List of fields that can be updated with update_profile."""
 
     @staticmethod
+    def get_by_nickname(nickname):
+        """Get a user by their nickname."""
+        try:
```

Comment:
I'm not convinced about wrapping SQLAlchemy exception. It would make sense if it was high-level API, but here we are still in model.